src/components/steps/StepBillingData.tsx

```diff
@@ -79,23 +84,38 @@ export default function StepBillingData({
     back!();
   };
 
+  // eslint-disable-next-line sonarjs/cognitive-complexity
   const validate = (values: Partial<BillingData>) =>
     Object.fromEntries(
       Object.entries({
         businessName: !values.businessName ? requiredError : undefined,
         registeredOffice: !values.registeredOffice ? requiredError : undefined,
-        taxCode: !values.taxCode ? requiredError : undefined,
-        vatNumber:
-          stepHistoryState.isTaxCodeNotEquals2PIVA && !values.vatNumber ? requiredError : undefined,
-        digitalAddress: !values.digitalAddress
+        taxCode: !values.taxCode
+          ? requiredError
+          : !stepHistoryState.isTaxCodeNotEquals2PIVA &&
+            values.taxCode &&
+            !onlyNumbersRegexp.test(values.taxCode)
+          ? t('stepBillingData.invalidFiscalCode')
+          : values.taxCode &&
+            stepHistoryState.isTaxCodeNotEquals2PIVA &&
+            !fiscalCodeRegexp.test(values.taxCode)
+          ? t('stepBillingData.invalidFiscalCode')
+          : undefined,
```

Comment:
In this way you will not test for the fiscalCode when te onlyNumberRegexp will fail
Thus this check is not correct!